Challenge1/src/agent.ts

```diff
@@ -1,78 +1,43 @@
-import {
-  BlockEvent,
-  Finding,
-  Initialize,
-  HandleBlock,
-  HealthCheck,
-  HandleTransaction,
-  HandleAlert,
-  AlertEvent,
-  TransactionEvent,
-  FindingSeverity,
-  FindingType,
-} from "forta-agent";
+import { Finding, FindingSeverity, FindingType, HandleTransaction, TransactionEvent, ethers } from "forta-agent";
 
-export const BOT_DEPLOYMENT_FUNCTION =
-  "function createAgent(uint256 agentId,address ,string metadata,uint256[] chainIds)";
-const NETHERMIND_ADDRESS = "0x88dC3a2284FA62e0027d6D6B1fCfDd2141a143b8".toLowerCase();
+const BOT_DEPLOYMENT_FUNCTION = "function createAgent(uint256 agentId,address ,string metadata,uint256[] chainIds)";
+// const BOT_UPDATE_FUNCTION = "function updateAgent(uint256 agentId,string metadata,uint256[] chainIds)";
+const BOT_UPDATE_FUNCTION = "function updateAgent(uint256,string,uint256[])";
+const NETHERMIND_ADDRESS = "0x88dC3a2284FA62e0027d6D6B1fCfDd2141a143b8";
 const FORTA_REGISTRY_ADDRESS = "0x61447385B019187daa48e91c55c02AF1F1f3F863";
-let findingsCount = 0;
 
-const handleTransaction: HandleTransaction = async (txEvent: TransactionEvent) => {
-  const findings: Finding[] = [];
-
-  // limiting this agent to emit only 5 findings so that the alert feed is not spammed
-  if (findingsCount >= 5) return findings;
-  // only run this agent on the Nethermind address
-  if (txEvent.from !== NETHERMIND_ADDRESS) return findings;
-
-  const callsToCreateAgent = txEvent.filterFunction(BOT_DEPLOYMENT_FUNCTION, FORTA_REGISTRY_ADDRESS);
-
-  callsToCreateAgent.forEach((call) => {
-    findings.push(
-      Finding.fromObject({
-        name: "Nethermind Bot Deployment",
-        description: `Nethermind deployed a new bot with ID: ${call.args.agentId}`,
-        alertId: "NEW-BOT-DEPLOYED",
-        severity: FindingSeverity.Low,
-        type: FindingType.Info,
-        metadata: {
-          agentId: call.args.agentId,
-        },
-      })
-    );
-    findingsCount++;
-  });
-  return findings;
+const provideHandleTransaction = (nethermindAddress: string, fortaRegistryAddress: string): HandleTransaction => {
+  const handleTransaction: HandleTransaction = async (txEvent: TransactionEvent) => {
+    const findings: Finding[] = [];
+
+    // Convert addresses to lowercase for comparison
+    if (txEvent.from.toLowerCase() !== nethermindAddress.toLowerCase()) return findings;
+    if (txEvent.to?.toLowerCase() !== fortaRegistryAddress.toLowerCase()) return findings;
+
+    const botEvents = txEvent.filterFunction([BOT_DEPLOYMENT_FUNCTION, BOT_UPDATE_FUNCTION], fortaRegistryAddress);
+
+    botEvents.forEach((event) => {
+      const isDeployment = event.name === "createAgent";
```

Comment:
Note that the are not "events" (they would be if you were using `.filterLog`), so I'd suggest renaming the variables here